src/parser/mod.rs

```diff
@@ -4083,6 +4083,13 @@ impl<'a> Parser<'a> {
             None
         };
 
+        // In BigQuery PARITION BY can be also defined after columns
+        let partitioned_by = if self.parse_keywords(&[Keyword::PARTITION, Keyword::BY]) {
+            Some(self.parse_expr()?)
```

Comment:
Went through code base and parsing `PARTITION BY` or `PARTITION` in parts like window specification are always parsing just expr. Keeping the same parsing.

Also not guarding it by dialect as whole `parse_create_table` is dialect independent.